tests/test_plot.py

```diff
@@ -194,6 +200,37 @@ def test_pairplot_outputs():
     )
 
 
+def test_pairplot_with_gmm_inputs():
+    X = np.random.rand(15, 3)
+    gmm = GaussianMixture(n_components=3, covariance_type="full").fit(X)
+    labels = ["A"] * 5 + ["B"] * 5 + ["C"] * 5
+    # test data
+    with pytest.raises(TypeError):
+        pairplot_with_gmm(X="test", gmm=gmm)
+
+    with pytest.raises(ValueError):
+        pairplot_with_gmm(X=X, gmm=gmm, labels=["A"])
+
+    with pytest.raises(NameError):
+        pairplot_with_gmm(X, gmm=None)
+
+
+def test_pairplot_with_gmm_outputs():
+    X = np.random.rand(15, 3)
+    gmm = GaussianMixture(n_components=3, covariance_type="full").fit(X)
+    labels = ["A"] * 5 + ["B"] * 5 + ["C"] * 5
+    cluster_palette = {0: "red", 1: "blue", 2: "green"}
+    label_palette = {"A": "red", "B": "blue", "C": "green"}
+    fig = pairplot_with_gmm(X, gmm)
+    fig = pairplot_with_gmm(
+        X,
+        gmm,
+        labels=labels,
+        cluster_palette=cluster_palette,
+        label_palette=label_palette,
+    )
+
```

Comment:
in case you haven't seen this stuff

https://github.com/matplotlib/pytest-mpl
https://matplotlib.org/3.1.1/api/testing_api.html